File: zero-business/zero-main-business/zero-main-domain-business/src/main/scala/com/zjj/main/domain/user/UserQueryModel.scala
package com.zjj.main.domain.user

import org.jmolecules.architecture.cqrs.QueryModel
import org.springframework.stereotype.Service

/**
 * @author zengJiaJun
 * @crateTime 2024年12月25日 21:13
 * @version 1.0
 */
@Service
@QueryModel
class UserQueryModel(userRepository: UserRepository) {



  def findByUsername(username: String): io.vavr.control.Option[UserAgg] = {
    userRepository.findByUserName(username)
  }

}

File: zero-business/zero-main-business/zero-main-domain-business/src/main/java/com/zjj/main/domain/user/UserQueryModel.java
package com.zjj.main.domain.user;

import com.zjj.autoconfigure.component.security.UserProfile;
import com.zjj.main.domain.user.model.UserRecord;
import io.vavr.control.Option;
import lombok.RequiredArgsConstructor;
import org.jmolecules.architecture.cqrs.QueryModel;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;



/**
 * @author zengJiaJun
 * @version 1.0
 * @crateTime 2025年03月24日 16:46
 */
@Service
@QueryModel
@RequiredArgsConstructor(onConstructor = @__(@Autowired))
public class UserQueryModel {

    private final UserReadRepository userRepository;

    public Option<UserRecord> findByUsername(String username) {
        return userRepository.findByUsername(username);
    }

    public UserProfile findProfileByUsername(String username) {
        return userRepository.findProfileByUsername(username);
    }
}
